### PR TITLE
Configurable build resources - changes

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

Changes proposed in this pull request:

Configurable build resources:
- Add new property in Function's CRD -> `buildResources`
- Add presets for `resources`, `buildResources` and `replicas`
- Update CBS with new `buildResources` field

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/9227

### CODE REVIEW

**Review by aerfio — CHANGES_REQUESTED** [review]

File: components/console-backend-service/go.mod [hunk 1]
```diff
@@ -58,6 +57,8 @@ require (
 replace (
 	github.com/aws/aws-sdk-go => github.com/aws/aws-sdk-go v1.34.1
 	github.com/coreos/etcd => github.com/coreos/etcd v3.3.24+incompatible
+	github.com/kyma-project/kyma/components/function-controller => github.com/magicmatatjahu/kyma/components/function-controller v0.0.0-20200923082203-ab74f68958f1
 	golang.org/x/text => golang.org/x/text v0.3.3
 	k8s.io/client-go => k8s.io/client-go v0.17.4
+	k8s.io/utils => k8s.io/utils v0.0.0-20191114184206-e782cd3c129f+incompatible
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply components/console-backend-service/go.mod:63]: `k8s.io/utils` versions from 2020* year (btw. it was fetched via changes to function-controller) has one of standard libs (as I remember `time`) of go in `1.13` version and cbs is in `1.12`, so in compilation time of tests I had an errors - only in tests.

**Review by aerfio — APPROVED** [review]

**Review by Wawrzyn321 — APPROVED** [review]

**Pull request merged**